### DISCUSSION TITLE
"Preview" mode support?

### DISCUSSION THREAD

**Discussion by AaronBeaudoin** [directus/directus#12837] [Archive]
A major need when using Directus as a CMS for a website is the ability to preview changes without necessary going live with them.

The only way to do this with Directus seems to be to have a staging instance which via some trigger can be cloned over to a production instance when the user desires. This has a few issues.

- It requires two full instances of Directus and an associated database to be running all the time (and two website instances, unless the preview is triggered via a query parameter or something).
- Any user who triggers the "push" to the production instance will clone not only their own changes but the entire database, even if some other users changes may not be ready to go live. Unless, of course, some custom logic is implemented to look at the changes history and only push the current user's changes.

What is the recommended approach right now, and is there any plan to better support this workflow in the future? I would imagine being able to preview changes before exposing them to the public is a pretty common business requirement.

**Comment by rijkvanzanten** [comment]
I believe what you're referring too is very similar to the ongoing discussion in https://github.com/directus/directus/discussions/2975. I'd love to hear any and all thoughts you might have on the flow over there 🙂

**Reply by AaronBeaudoin** [reply]
Ah, sorry I should have seen that thread.

**Reply by rijkvanzanten** [reply]
No problem! Hard to find sometimes in the long list of good ideas 🙂

**Comment by benhaynes** [comment]
There are other ways you can preview content too — basically you need to authenticate on your front-end experience and then don't filter out draft content. We'll be writing an article on this soon.

**Reply by AaronBeaudoin** [reply]
Has this article landed?
